tests/test_delay.js

```diff
@@ -312,4 +312,68 @@ describe('`delayConnection()`', () => {
       100
     )
   })
+
+  it('emits a timeout - with setTimeout', done => {
+    nock('http://example.test').get('/').delayConnection(10000).reply(200, 'OK')
+
+    const onEnd = sinon.spy()
+
+    const req = http.request('http://example.test', res => {
+      res.once('end', onEnd)
+    })
+
+    req.setTimeout(5000, () => {
```

Comment:
It seems like this test would add a relatively long runtime to the tests. Could these timers be shortened a bit?

Reply:
This long value is never waited. In fact the `1000` passed to `delayConnection` doesn't wait either, so this test passes only after a handful of ms.

Reply:
Hmm, I'm pretty confused about that, at least thinking from the caller's perspective. 🤔

Reply:
I guess this is covered in the changes to the readme. It makes sense, though I think it's probably worth expanding it more in the docs.

Reply:
I added a few comments to the docs, which I think is probably the place this should be explained. (Or possibly we could briefly refer folks to the readme here, though I'm not sure that is really necessary.)